launcher/ui/widgets/InfoFrame.cpp

```diff
@@ -119,39 +116,49 @@ void InfoFrame::updateWithResourcePack(ResourcePack& resource_pack)
     // TODO: Make the same logic for font formatting too.
     // TODO: Wrap links inside <a> tags
 
-    auto description = resource_pack.description();
-
-    QString description_parsed("<html>");
+    QString html("<html>");
     bool in_div = false;
 
-    auto desc_it = description.constBegin();
-    while (desc_it != description.constEnd()) {
-        if (*desc_it == u'§') {
+    auto it = input.constBegin();
+    while (it != input.constEnd()) {
+        if (*it == u'§') {
             if (in_div)
-                description_parsed += "</span>";
+                html += "</span>";
 
-            auto const& num = *(++desc_it);
-            description_parsed += QString("<span style=\"color: %1;\">").arg(s_value_to_color.constFind(num).value());
+            auto const& num = *(++it);
+            html += QString("<span style=\"color: %1;\">").arg(s_value_to_color.constFind(num).value());
 
             in_div = true;
 
-            desc_it++;
+            it++;
         }
 
-        description_parsed += *desc_it;
-        desc_it++;
+        html += *it;
+        it++;
     }
 
     if (in_div)
-        description_parsed += "</span>";
-    description_parsed += "</html>";
+        html += "</span>";
+    html += "</html>";
 
-    description_parsed.replace("\n", "<br>");
+    html.replace("\n", "<br>");
+    return html;
+}
 
-    setDescription(description_parsed);
+void InfoFrame::updateWithResourcePack(ResourcePack& resource_pack)
+{
+    setName(resource_pack.name());
+    setDescription(renderColorCodes(resource_pack.description()));
     setImage(resource_pack.image({64, 64}));
 }
 
+void InfoFrame::updateWithTexturePack(TexturePack& texture_pack)
+{
+    setName(texture_pack.name());
+    setDescription(texture_pack.description());
```

Comment:
```suggestion
    setDescription(renderColorCodes(texture_pack.description()));
```

as i've said in the comment, it looks like color codes are accepted in texture pack descriptions, otherwise minecraft wouldn't parse it in-game ig (which it does).

Reply:
Oh. I thought I already did that and was confused about your comment xD

Reply:
The whole reason, why i split it out into a new function was, so that I can reuse it for texture packs :D

Reply:
makes sense, and it looks better with your changes too =D